File: backend/src/main/java/com/cbp7/payment/service/PaymentService.java
package com.cbp7.payment.service;

import com.cbp7.identity.auth.entity.User;
import com.cbp7.payment.dto.request.CreatePaymentRequest;
import com.cbp7.payment.dto.response.PaymentDetailResponse;
import com.cbp7.payment.dto.response.PaymentResponse;
import com.cbp7.payment.dto.response.PaymentStatusResponse;
import com.cbp7.payment.dto.response.PhonePePaymentResponse;
import com.cbp7.payment.dto.response.PhonePeStatusDetailsResponse;

public interface PaymentService {

    PaymentResponse createPayment(User user, CreatePaymentRequest request);

    PhonePePaymentResponse initiatePhonePePayment(User user);

    PaymentDetailResponse getMyPayment(User user);

    PaymentStatusResponse verifyAndGetPaymentStatus(User user, String transactionId);
    
    PhonePeStatusDetailsResponse verifyAndGetStatusDetails(User user, String transactionId);
}

File: frontend/src/services/paymentService.ts
import { api } from "@/utils/api"
import {
  PaymentDetailResponse,
  PaymentResponse,
  PhonePePaymentResponse,
  PaymentStatusResponse,
} from "@/types/payment"

export const paymentService = {
  getMyPayment: () => api.get<PaymentDetailResponse>("/api/v1/payment/me"),
  initiatePhonePe: () => api.post<PhonePePaymentResponse>("/api/v1/payment/phonepe/initiate"),
  createPayment: (amount: number, registrationId: string) =>
    api.post<PaymentResponse>("/api/v1/payment/create", { amount, registrationId }),
  getPaymentStatus: (transactionId: string) =>
    api.get<PaymentStatusResponse>(`/api/v1/payment/${transactionId}/status`),
}
